src/streaming-client.ts

```diff
@@ -0,0 +1,117 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import * as events from 'events';
+import { Duplex, PassThrough } from 'stream';
+
+import AudioConfig from './models/streaming/AudioConfig';
+import BufferedDuplex from './models/streaming/BufferedDuplex';
+import { 
+    StreamingResponse, 
+    StreamingHypothesis, 
+    StreamingConnected 
+} from './models/streaming/StreamingResponses';
+
+export class RevAiStreamingClient extends events.EventEmitter {
+    baseUrl: string;
+    client: sockets.WebSocketClient;
+    private accessToken: string;
+    private config: AudioConfig;
+    private requests: PassThrough;
+    private responses: PassThrough;
+    constructor(accessToken: string, config: AudioConfig, version = 'v1alpha'){
+        super();
+        this.accessToken = accessToken;
+        this.config = config;
+        this.baseUrl = `wss://api.rev.ai/speechtotext/${version}/stream?` +
+            `access_token=${accessToken}` +
+            `&content_type=${config.getContentTypeString()}`;
+        this.requests = new PassThrough();
+        this.responses = new PassThrough({ objectMode: true });
+        this.client = new sockets.client();
+    }
+
+    start(): Duplex {
+        this.setUpClient();
+        this.client.connect(this.baseUrl);
+        return new BufferedDuplex(this.requests, this.responses, { readableObjectMode: true });
+    }
+
+    end(): void {
+        this.client.abort();
+        this.requests.end("EOS");
+        this.responses.push(null);
+    }
+
+    private setUpClient(): void {
+        var self = this;
+        this.client.on('httpResponse', function(response, client) {
```

Comment:
minor, I would move each of these `this.client.on` into separate private helper functions. 

so `setupHttpResponseHandler`, `setupConnectionFailedHandler`, `setupConnectHandler`, etc